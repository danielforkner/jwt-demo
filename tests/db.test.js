require('dotenv').config();
const client = require('../db/client');
const { buildDb } = require('../db/seed');
const { createUser, getUserById } = require('../db');

describe('Testing the database', () => {
  beforeAll(async () => {
    await client.connect();
    await buildDb();
  });
  afterAll(async () => {
    await client.end();
  });

  describe('Users', () => {
    const userToCreate = { username: 'jest', password: '1234' };
    const badUserToCreate = { username: 'Bob', password: null };
    let createdUser;
    test('Create User: returns an object', async () => {
      createdUser = await createUser(userToCreate);
      expect(typeof createdUser).toBe('object');
    });
    test('Create User: returns the same user just created', async () => {
      expect(createdUser.username).toBe('jest');
      expect(createdUser.password).toBe('1234');
    });
    test('Get User: can get the user by its id', async () => {
      const identicalUser = await getUserById(createdUser.id);
      expect(identicalUser).toEqual(createdUser);
    });
    test('Create User: prevents null password', async () => {
      await expect(
        createUser(badUserToCreate)
      ).rejects.toThrowErrorMatchingSnapshot();
    });
    test.todo('Create User: prevents empty string password');
  });
});
