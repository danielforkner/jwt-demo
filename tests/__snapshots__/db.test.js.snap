// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the database Users Create User: prevents null password 1`] = `"null value in column "password" of relation "users" violates not-null constraint"`;
